src/lock.js

```diff
@@ -68,4 +69,42 @@ class Lock {
   }
 }
 
-export const lock = new Lock();
+class GlobalLock {
+  constructor() {
+    this.wallets = {};
+    this.globalLock = new Lock();
+  }
+
+  start(walletId) {
```

Comment:
```suggestion
  _start(walletId) {
```

Reply:
suggestion (non-blocking): this method shouldn't be used outside of the class.

Reply:
start method was removed